eng/common/TestResources/New-TestResources.ps1

```diff
@@ -154,14 +154,21 @@ if ($ProvisionerApplicationId) {
     $subscriptionArgs = if ($SubscriptionId) {
         @{SubscriptionId = $SubscriptionId}
     }
+    else {
+        ""
```

Comment:
It is, unfortunately, necessary.  It was bombing out when `-WhatIf` was passed due to an interpreted `$null`.    I'll give it a look and see how it likes the empty hashtable.   I suspect that should satisfy things and agree it would be a better approach.